### PR TITLE
add some APIs and empty methods

### PR DESCRIPTION
**Pull request by process852**
1. 增加了 `oneflow._C._nn._parse_to` 接口，具体参看 https://github.com/Oneflow-Inc/OneCloud/issues/185#issuecomment-1430684732 以及 https://github.com/Oneflow-Inc/OneCloud/issues/185#issuecomment-1435821518
2. 增加了 `oneflow._C._log_api_usage_once` 接口 https://github.com/Oneflow-Inc/OneCloud/issues/185#issuecomment-1429131625
3. 增加了 `oneflow.device()` 允许接收 `oneflow.device` 参数与 pytorch 对齐
image

4. 调整了 `GradScaler` 接口的位置， oneflow 中在 `oneflow.amp` 实现了对应接口，而 pytorch 则是在 `torch.cuda.amp` 实现
5. 调整了 `swa_utils` 接口的导入方式匹配 pytorch 的导入方式，原先的方式不支持`from oneflow.optim.swa_utils import SWALR` 导入
6. 新增了一些空接口用于匹配框架迁移过程导入包时的报错行为(是否合并进入主仓库还需讨论)，建立空接口的部分原因是这些接口 API 大多是在迁移过程中导入包的时候用到，实际运行过程用到机会并不多，而直接开发的性价比有比较低，故考虑暂时使用空接口。
    - `oneflow.backends.cudnn.benchmark` 默认为 False
    - `oneflow.cuda.Event`
    - `oneflow.distributed.is_available` 默认为 False, oneflow 有自己的分布式流程
    - `oneflow.linalg.solve`
    - `oneflow.nn.DataParallel`
    - `oneflow.optim.LBFGS`
    - `oneflow.profiler.tensorboard_trace_handler`
    - `oneflow.profiler.ProfilerAction`

### CODE REVIEW

**Comment by process852** [comment]
https://github.com/Oneflow-Inc/oneflow/pull/9877 与本 PR 内容完全一致，关闭之前那个 PR 的原因是上周 26 号服务器坏了之后，我在 27 号新开了账户，但是设置的 git 账户没有关联到 github 账户，那个账户和我现有的账户冲突，所以我在本地修改后重新开了一个，但后来发现直接换名称就可以，这一步也没必要。。。

下面我贴的是大老师的评论和需要修改的地方，方便有需要时查看
https://github.com/Oneflow-Inc/oneflow/pull/9877#discussion_r1111408545

新增 commit 主要包括：
1. 为空方法增加了 warning 警告
2. 修改了 `oneflow._C._nn._parse_to` 实现逻辑

File: python/oneflow/autograd/profiler.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+limitations under the License.
+"""
+from oneflow.profiler.profiler import profile
+
+
+def record_function():
+    raise NotImplementedError()
+
+
+class emit_nvtx(object):
```
Comment (daquexian) [inline python/oneflow/autograd/profiler.py:23]:
```suggestion
class emit_nvtx():
```
object 可以省略

File: python/oneflow/_C/__init__.py [hunk 2]
```diff
@@ -14,7 +14,16 @@
...
 
 
 def allclose(input, other, atol=1e-08, rtol=1e-05, equal_nan=False):
     return isclose(input, other, atol, rtol, equal_nan).all().item()
+
+
+def _log_api_usage_once(event):
+    warnings.warn(
+        "The _log_api_usage_once API only support training mode, \
+                  debugging mode in pytorch is temporarily not supported."
```
Comment (daquexian) [inline python/oneflow/_C/__init__.py:28]: 这是什么意思

File: python/oneflow/_C/_nn.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            else:
+                flag = True
+
+    if flag and index != -1:
+        non_blocking_param = new_args.pop(index)
+        if non_blocking_param:
+            raise ValueError("oneflow temporarily supports 'non_blocking is False' ")
+        result = flow._C.to(*new_args)
+    else:
+        result = flow._C.to(*new_args)
```
Comment (daquexian) [inline python/oneflow/_C/_nn.py:88]: 😂 我昨天不是这个意思，不过我又看了一下昨天是我搞错了，这里我来改一下吧

File: python/oneflow/_C/__init__.py (hunk 2 above)
Comment (process852) [reply python/oneflow/_C/__init__.py:28]: 这个部分 torch 中用的一个环境变量 `PYTORCH_API_USAGE_STDERR` 来控制处于调式模式时会打印事件的名称，而不是调试的时候就是什么也不做的空函数。然后我打印这个警告就是说明，oneflow 目前仅仅支持什么也不做的情况，而调试的时候打印输出事件暂时不支持的。

image

File: python/oneflow/autograd/profiler.py (hunk 1 above)
Comment (process852) [reply python/oneflow/autograd/profiler.py:23]: 好的

File: python/oneflow/_C/__init__.py (hunk 2 above)
Comment (daquexian) [reply python/oneflow/_C/__init__.py:28]: 哦哦，training mode 是自己取的名字吗，不太贴切。改成类似这样吧：

```python
if os.getenv("PYTORCH_API_USAGE_STDERR") is not None:
  warnings.warn("_log_api_usage_once is not implemented in oneflow")

Comment (process852) [reply python/oneflow/_C/__init__.py:28]: 我自己临时取的。。。，我改一下这个警告提示

File: python/oneflow/_C/_nn.py (hunk 3 above)
Comment (daquexian) [inline python/oneflow/_C/_nn.py:88]: 要不先不考虑 memory_format 和 non_blocking 了，`_parse_to` 作为 PyTorch 的内部函数除了 lightning 之外估计没有什么库会用，写太长的逻辑不明智。用一行这个代替算了：
```suggestion
    result = flow.tensor([]).to(*args, **kwargs)
```

File: python/oneflow/__init__.py [hunk 4]
```diff
@@ -71,6 +72,11 @@ def is_deprecated(func_or_class):
     )
 
 
+def use_deterministic_algorithms(mode, *, warn_only=False):
+    # register a empty method
+    warnings.warn("Oneflow temporarily not supports use_deterministic_algorithms.")
```
Comment (daquexian) [inline python/oneflow/__init__.py:77]:
```suggestion
    warnings.warn("Oneflow temporarily does not support use_deterministic_algorithms.")
```

File: python/oneflow/_C/_nn.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+def _parse_to(*args, **kwargs):
+    new_args = list()
+    # If device is single int, replace it with flow.device("cuda:{device}")
+    if len(args) > 0 and isinstance(args[0], int):
+        new_args.append(flow.device(f"cuda:{args[0]}"))
+        for i in range(1, len(args)):
+            new_args.append(args[i])
+    else:
+        new_args = args
```
Comment (daquexian) [inline python/oneflow/_C/_nn.py:60]: 这个没必要了呀，在 tensor.to 里已经写过这个逻辑了

Comment (process852) [reply python/oneflow/_C/_nn.py]: 嗯嗯，是的。我弄混了, `_to` 和 `to` 的逻辑是一样的

**Review by daquexian — APPROVED** [review]

**Review by doombeaker — APPROVED** [review]
